core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -135,7 +137,15 @@ public CloseableIterable<FileScanTask> planFiles() {
             planningDuration.stop();
             ScanReport scanReport =
                 ImmutableScanReport.builder()
-                    .projection(schema())
+                    .schemaId(schema().schemaId())
+                    .projectedFieldIds(
+                        schema().columns().stream()
+                            .map(NestedField::fieldId)
+                            .collect(Collectors.toList()))
```

Comment:
This needs to also include the nested IDs by calling `TypeUtil.getProjectedIds`.

Reply:
done